File: ToolGood.Algorithm/MathNet/Statistics/Statistics.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ToolGood.Algorithm.MathNet.Numerics.Statistics
{
    public static class Statistics
    {
        public static double QuantileCustom(this IEnumerable<double> data, double tau, QuantileDefinition definition)
        {
            double[] array = data.ToArray();
            return ArrayStatistics.QuantileCustomInplace(array, tau, definition);
        }
        public static double QuantileRank(this IEnumerable<double> data, double x, RankDefinition definition = RankDefinition.Default)
        {
            double[] array = data.ToArray();
            Array.Sort(array);
            return SortedArrayStatistics.QuantileRank(array, x, definition);
        }


    }
}

File: java/toolgood.algorithm/src/main/java/toolgood/algorithm/mathNet/Statistics/Statistics.java
package toolgood.algorithm.mathNet.Statistics;

import java.awt.List;

public class Statistics {
    public static double QuantileCustom(final List<Double> data, final double tau,
            final QuantileDefinition definition) {
        double[] array = new double[data.size()];
        for (int i = 0; i < data.size(); i++) {
            array[i]=data.get(i);
        }
        return ArrayStatistics.QuantileCustomInplace(array, tau, definition);
    }

    public static double QuantileRank(final List<Double> data, final double x) {
        List<Double> dt=ShellSort(data);
        double[] array = new double[dt.size()];
        for (int i = 0; i < dt.size(); i++) {
            array[i]=dt.get(i);
        }
        return SortedArrayStatistics.QuantileRank(array, x);
    }

    private static List<Double> ShellSort(List<Double> array) {
        int len = array.size();
        double temp;
        int gap = len / 2;
        while (gap > 0) {
            for (int i = gap; i < len; i++) {
                temp = array.get(i);
                int preIndex = i - gap;
                while (preIndex >= 0 && array.get(preIndex) > temp) {
                    array.set(preIndex + gap, array.get(preIndex));
                    // array[preIndex + gap] = array[preIndex];
                    preIndex -= gap;
                }
                array.set(preIndex + gap, temp);
                // array[preIndex + gap] = temp;
            }
            gap /= 2;
        }
        return array;
    }
}